packages/core-contracts/contracts/utils/ERC20.sol

```diff
@@ -0,0 +1,101 @@
+//SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "../interfaces/IERC20.sol";
+
+abstract contract ERC20 is IERC20 {
+    string public override name;
+
+    string public override symbol;
+
+    uint8 public immutable override decimals;
```

Comment:
Isn't it weird to make this immutable and not the others because Solidity does not allow it?